src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -513,8 +514,9 @@ public string ProcessName
         {
             get
             {
-                EnsureState(State.HaveProcessInfo);
-                return _processInfo!.ProcessName;
+                EnsureState(State.HaveProcessName);
```

Comment:
ProcessName property can be moved to Windows and Unix partials. It will avoid GetProcessName overhead on Unix, and we will only cache `_processInfo` (to load the info just once) as before.

Reply:
Sorry @am11 - could you elaborate? Do you mean adding these attributes on this property?
```
        [SupportedOSPlatform("windows")]
        [SupportedOSPlatform("linux")]
```

Reply:
* In `Process.Unix.cs`: original `ProcessName` as it was on main branch.
* In `Process.Windows.cs`: this modified `ProcessName` property from your PR.
* In `Process.UnknownUnix.cs`: `public string ProcessName => throw new PlatformNotSupportedException();`

This way, we won't need to introduce new method in `ProcessManaged.Unix.cs`.